lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -339,13 +339,15 @@ def workflow_dict(self, trans: GalaxyWebTransaction, workflow_id, **kwd):
         style = kwd.get("style", "export")
         download_format = kwd.get("format")
         version = kwd.get("version")
+        instance = util.string_as_bool(kwd.get("instance", "false"))
+        instance_id = self.decode_id(workflow_id) if instance else None
```

Comment:
You can do that unconditionally on line 337, before `self.__get_stored_accessible_workflow`. id decoding should happen as early as possible, and `__get_stored_accessible_workflow` works fine with decoded ids.

Reply:
Ok, I can do this above L 337:
```python
        instance = util.string_as_bool(kwd.get("instance", "false"))
        workflow_id = self.decode_id(workflow_id)
        instance_id = workflow_id if instance else None
```